tritium-metrics/src/main/java/com/palantir/tritium/metrics/TaggedMetricsScheduledExecutorService.java

```diff
@@ -35,52 +34,65 @@
 
     private final Meter submitted;
     private final Counter running;
-    private final Meter completed;
     private final Timer duration;
 
-    private final Meter scheduledOnce;
-    private final Meter scheduledRepetitively;
     private final Counter scheduledOverrun;
-    private final Histogram scheduledPercentOfPeriod;
 
     TaggedMetricsScheduledExecutorService(ScheduledExecutorService delegate, ExecutorMetrics metrics, String name) {
         this.delegate = delegate;
         this.name = name;
 
         this.submitted = metrics.submitted(name);
         this.running = metrics.running(name);
-        this.completed = metrics.completed(name);
         this.duration = metrics.duration(name);
 
-        this.scheduledOnce = metrics.scheduledOnce(name);
-        this.scheduledRepetitively = metrics.scheduledRepetitively(name);
         this.scheduledOverrun = metrics.scheduledOverrun(name);
-        this.scheduledPercentOfPeriod = metrics.scheduledPercentOfPeriod(name);
     }
 
     @Override
     public ScheduledFuture<?> schedule(Runnable task, long delay, TimeUnit unit) {
-        scheduledOnce.mark();
-        return delegate.schedule(new TaggedMetricsRunnable(task), delay, unit);
+        ScheduledFuture<?> future = delegate.schedule(new TaggedMetricsRunnable(task), delay, unit);
+        // RejectedExecutionException should prevent 'submitted' from being incremented.
+        // This means a wrapped same-thread executor will produce delayed 'submitted' values,
+        // however the results will work as expected for the more common cases in which
+        // either a queue is full, or the delegate has shut down.
+        submitted.mark();
```

Comment:
This definitely isn't right.`submitted` should be updated when we submit the task to the executor - not when we schedule the tasks. Otherwise we'll be double counting submissions.

It's also weird to mark it here because it diverges from the meaning of `submitted` in non-scheduled executors.

Reply:
I'm not sure I follow, this matches how we mark `submitted` in non-scheduled executors. If I schedule a runnable to execute in 1 minute, the `submitted` metric works exactly like an instrumented fixed thread pool executor where the task sits in a queue for 1 minute.

> Otherwise we'll be double counting submissions.

This would cause tests to fail unless I've misunderstood your concern